pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -187,38 +186,18 @@ func getRemoteImage(image string, kubeclient kubernetes.Interface, build *buildv
 	if err != nil {
 		return nil, fmt.Errorf("Failed to parse image %s: %v", image, err)
 	}
-	var kc authn.Keychain
+
 	serviceAccountName := build.Spec.ServiceAccountName
-	if serviceAccountName != "" && serviceAccountName != "default" {
-		sa, err := kubeclient.CoreV1().ServiceAccounts(build.Namespace).Get(serviceAccountName, metav1.GetOptions{})
-		if err != nil {
-			return nil, fmt.Errorf("Failed to get service account %s: %v", serviceAccountName, err)
-		}
-		if len(sa.ImagePullSecrets) == 0 {
-			return nil, fmt.Errorf("No ImagePullSecret for service account %s: %v", serviceAccountName, err)
-		}
-		for _, secret := range sa.ImagePullSecrets {
-			_, err = kubeclient.CoreV1().Secrets(build.Namespace).Get(secret.Name, metav1.GetOptions{})
-			if err != nil {
-				return nil, fmt.Errorf("Failed to get ImagePullSecret for service account %s: %v", serviceAccountName, err)
-			}
 
-			kc, err = k8schain.New(kubeclient, k8schain.Options{
-				ImagePullSecrets: []string{secret.Name},
-				Namespace:        build.Namespace,
-			})
-			if err != nil {
-				return nil, fmt.Errorf("Failed to create k8schain: %v", err)
-			}
-		}
-	} else {
-		kc, err = k8schain.New(kubeclient, k8schain.Options{
-			Namespace:          build.Namespace,
-			ServiceAccountName: "default",
-		})
-		if err != nil {
-			return nil, fmt.Errorf("Failed to create k8schain: %v", err)
-		}
+	if serviceAccountName == "" {
```

Comment:
Library does this piece of code as well. Pass the `serviceAccountName ` (empty or not) to k8schain library

Reply:
Whoops, I actually had that thought and then it slipped my mind. Updated.